python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -0,0 +1,240 @@
+    for each target/source combination, as a matrix of lists.
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure for which to measure crosstalk.
+    threshold : `float`
+        Lower limit on pixels for which we measure crosstalk.
+    badPixels : `list` of `str`
+        Mask planes indicating a pixel is bad.
+
+    Returns
+    -------
+    ratios : `list` of `list` of `list` of `float`
+        A matrix of lists of ratios. ``ratios[i][j]`` is a list of
+        the fraction of the ``j``-th amp present on the ``i``-th amp.
+    """
+    mi = exposure.getMaskedImage()
+    FootprintSet(mi, Threshold(threshold), "DETECTED")
+    detected = mi.getMask().getPlaneBitMask("DETECTED")
+    bad = mi.getMask().getPlaneBitMask(badPixels)
+    bg = calculateBackground(mi, badPixels + ["DETECTED"])
+
+    ccd = exposure.getDetector()
+
+    ratios = [[[] for iAmp in ccd] for jAmp in ccd]
+
+    for ii, iAmp in enumerate(ccd):
+        iImage = mi.Factory(mi, iAmp.getBBox())
+        iMask = iImage.getMask().getArray()
+        select = (iMask & detected > 0) & (iMask & bad == 0) & np.isfinite(iImage.getImage().getArray())
+        for jj, jAmp in enumerate(ccd):
+            if ii == jj:
+                continue
+            jImage = extractAmp(mi.getImage(), jAmp, iAmp.getReadoutCorner())
+            pixels = (jImage.getArray()[select] - bg)/iImage.getImage().getArray()[select]
+            ratios[jj][ii] += pixels.tolist()
+
+    return ratios
+
+
+def measureCrosstalkCoefficients(ratios, rejIter=3, rejSigma=2.0):
+    """Measure crosstalk coefficients from the ratios
+
+    Given a list of ratios for each target/source amp combination,
+    we measure a robust mean and error.
+
+    Parameters
+    ----------
+    ratios : `list` of `list` of `list` of `float`
+        Matrix of lists of ratios.
+    rejIter : `int`
+        Number of rejection iterations.
+    rejSigma : `float`
+        Rejection threshold (sigma).
+
+    Returns
+    -------
+    coeff : `numpy.ndarray`
+        Crosstalk coefficients.
+    coeffErr : `numpy.ndarray`
+        Crosstalk coefficient errors.
+    coeffNum : `numpy.ndarray`
+        Number of pixels for each measurement.
+    """
+    numAmps = len(ratios)
+    assert all(len(rr) == numAmps for rr in ratios)
+
+    coeff = np.zeros((numAmps, numAmps))
+    coeffErr = np.zeros((numAmps, numAmps))
+    coeffNum = np.zeros((numAmps, numAmps), dtype=int)
+
+    for ii, jj in itertools.product(range(numAmps), range(numAmps)):
+        if ii == jj:
+            continue
+        values = np.array(ratios[ii][jj])
+        values = values[np.abs(values) < 1.0]  # Discard unreasonable values
+        for rej in range(rejIter):
+            lo, med, hi = np.percentile(values, [25.0, 50.0, 75.0])
+            sigma = 0.741*(hi - lo)
+            good = np.abs(values - med) < rejSigma*sigma
+            print(ii, jj, rej, med, sigma, len(values), good.sum())
```

Comment:
print statement here isn't going to be helpful for debugging: please remove it, or turn it into a `log.debug()`, with labels for the various bits.